src/lib/auth/refresh-access-token.ts

```diff
@@ -0,0 +1,51 @@
+import { JWT } from 'next-auth/jwt'
+
+// TODO: validate that the refresh token flow works as implemented
+
+/**
+ * Takes a token, and returns a new token with updated
+ * `accessToken` and `accessTokenExpires`. If an error occurs,
+ * returns the old token and an error property
+ *
+ * Adapted from: https://next-auth.js.org/tutorials/refresh-token-rotation#server-side
+ */
+export default async function refreshAccessToken(token: JWT) {
+	try {
+		const url =
+			__config.dev_dot.auth.idp_url +
+			'/oauth2/token?' +
+			new URLSearchParams({
+				client_id: process.env.AUTH_CLIENT_ID,
+				client_secret: process.env.AUTH_CLIENT_SECRET,
+				grant_type: 'refresh_token',
+				refresh_token: token.refreshToken as string,
+			})
+
+		const response = await fetch(url, {
+			headers: {
+				'Content-Type': 'application/x-www-form-urlencoded',
+			},
+			method: 'POST',
+		})
```

Comment:
I know you're following the example from the docs, but I can't recall ever seeing an OAuth `POST` request with parameters in the URL. If this flow doesn't work, that's probably the first thing I'd change.